File: typescript/day3/solution1.ts
import { input } from "./input"

console.log(input.length);

let x=0
let y=0
let index = 0
let list = []
while(index < input.length){
    if (input.charAt(index) == "^"){
        y = y + 1
    }
    if (input.charAt(index) == "v"){
        y = y - 1
    }
    if (input.charAt(index) == ">"){
        x = x + 1
    }
    if (input.charAt(index) == "<"){
        x = x - 1
    }
    let coordinaat = "c" + x + "r" + y
    if (list.indexOf(coordinaat) == -1){
        list.push(coordinaat)
    }
    index = index + 1
}

console.log(list.length);

File: 2015/day6/solution1.py
import input
newInput=input.input.split("\n")
import numpy
matrix=numpy.zeros((1000,1000))
import re
index=0
posities=list()
while index<len(newInput):
    instruction=newInput[index]
    positie=re.findall(r"(\d+)",instruction)
    actie=re.findall(r"(\w+) \d",instruction)
    posities.append(positie)
    positie.append(actie[0])
    index=index+1
print(posities)
index=0
while index<len(posities):
    instruction=posities[index]
    for rij in range(int(instruction[0]),int(instruction[2])+1):
        for kolom in range(int(instruction[1]),int(instruction[3])+1):
            if instruction[4]=="on":
                matrix[rij][kolom]=1
            elif instruction[4]=="off":
                matrix[rij][kolom]=0
            elif instruction[4]=="toggle":
                if matrix[rij][kolom]==0:
                    matrix[rij][kolom]=1
                else:
                    matrix[rij][kolom]=0
            else:
                raise "problem"
    index=index+1
lightson=numpy.count_nonzero(matrix)
print(lightson)

